src/snowflake/connector/cursor.py

```diff
@@ -1208,15 +1208,23 @@ def _log_telemetry_job_data(
         self, telemetry_field: TelemetryField, value: Any
     ) -> None:
         """Builds an instance of TelemetryData with the given field and logs it."""
-        obj = {
-            "type": telemetry_field.value,
-            "source": self._connection.application if self._connection else CLIENT_NAME,
-            "query_id": self._sfqid,
-            "value": int(value),
-        }
         ts = get_time_millis()
         try:
-            self._connection._log_telemetry(TelemetryData(obj, ts))
+            self._connection._log_telemetry(
+                TelemetryData(
+                    generate_telemetry_data(
+                        from_dict={
+                            TelemetryField.KEY_TYPE.value: telemetry_field.value,
+                            TelemetryField.KEY_SOURCE.value: self._connection.application
+                            if self._connection
+                            else CLIENT_NAME,
+                            TelemetryField.KEY_SFQID.value: self._sfqid,
```

Comment:
`KEY_SFQID`'s value is `QueryID` which is different than `query_id` which is technically a breaking change.

also I found that the naming pattern is quite inconsistent among those telemetry keys, some use underscore, some use camel style, some start with lower while some with upper case.

can we break them or not?

Reply:
I updated the naming style to be `UpperCamelCase` and constringfy them.

Also I condensed the keys a bit:
- ErrorNumber -> ErrorCode
- ErrorMessage -> ExceptionMessage
- Stacktrace -> ExceptionStacktrace

Reply:
We should keep this field as snake_case and most of these fields should be snake_case with the exception of ErrorNumber, ErrorMessage and Stacktrace. In the JDBC driver, this is also snake_case and we should keep them as such. JDBC implementation for reference: https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/jdbc/telemetry/TelemetryUtil.java

Originally, all telemetry messages were snake_case and then when Andong implemented OOB (Out of Band) Telemetry he made the fields camelCase. OOB Telemetry is telemetry that gets collected before we authenticate to a Snowflake deployment and is sent to an AWS queue. This is "internal" telemetry since it gets collected after we authenticate to AWS and gets sent to an internal API.

Reply:
thanks for the context, code updated